src/test/model_service_test.cpp

```diff
@@ -44,106 +44,194 @@ TEST(ModelService, config_reload) {
 }
 
-::grpc::Status test_PerformModelStatusRequest(ModelServiceImpl& s, tensorflow::serving::GetModelStatusRequest& req, tensorflow::serving::GetModelStatusResponse& res) {
+TEST(ModelService, single_version_model) {
+    ConstructorEnabledModelManager manager;
     auto config = DUMMY_MODEL_CONFIG;
-    ovms::ModelManager& manager = ovms::ModelManager::getInstance();
-    manager.reloadModelWithVersions(config);
-    spdlog::info("req={} res={}", req.DebugString(), res.DebugString());
-    ::grpc::Status ret = s.GetModelStatus(nullptr, &req, &res);
-    spdlog::info("returned grpc status: ok={} code={} msg='{}'", ret.ok(), ret.error_code(), ret.error_details());
-    return ret;
+    ASSERT_EQ(manager.reloadModelWithVersions(config), StatusCode::OK);
+    tensorflow::serving::GetModelStatusRequest req;
+    tensorflow::serving::GetModelStatusResponse res;
+
+    auto model_spec = req.mutable_model_spec();
+    model_spec->Clear();
+    model_spec->set_name("dummy");
+    model_spec->mutable_version()->set_value(1);  // existing version
+
+    ASSERT_EQ(GetModelStatusImpl::getModelStatus(&req, &res, manager), StatusCode::OK);
+
+    ASSERT_EQ(res.model_version_status_size(), 1);
+    ASSERT_EQ(res.model_version_status().begin()->state(), tensorflow::serving::ModelVersionStatus_State_AVAILABLE);
+    ASSERT_EQ(res.model_version_status().begin()->has_status(), true);
+    ASSERT_EQ(res.model_version_status().begin()->status().error_code(), tensorflow::error::OK);
+    ASSERT_EQ(res.model_version_status().begin()->status().error_message(), "OK");
 }
 
-TEST(ModelService, empty_request) {
-    test_LoadModels();
-    ModelServiceImpl s;
+static const char* pipelineOneDummyConfig = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy",
+                "target_device": "CPU",
+                "model_version_policy": {"all": {}},
+                "nireq": 1
+            }
+        }
+    ],
+    "pipeline_config_list": [
+        {
+            "name": "dummyPipeline",
+            "inputs": ["custom_dummy_input"],
+            "nodes": [
+                {
+                    "name": "dummyNode",
+                    "model_name": "dummy",
+                    "type": "DL model",
+                    "inputs": [
+                        {"b": {"node_name": "request",
+                               "data_item": "custom_dummy_input"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "a",
+                         "alias": "new_dummy_output"}
+                    ] 
+                }
+            ],
+            "outputs": [
+                {"custom_dummy_output": {"node_name": "dummyNode",
+                                         "data_item": "new_dummy_output"}
+                }
+            ]
+        }
+    ]
+})";
+
+TEST(ModelService, pipeline) {
```

Comment:
I think we should have 3 tests for pipelines:
1) Version 1 requested
2) version default requested
3) version eg. 2 requested

Reply:
Done